contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,944 @@
+        // the deployer shall call safeApproveAllTokens() to set necessary approvals
+        revert("Unsupported method");
+    }
+
+    /**
+     * @dev Approve the spending of all assets
+     */
+    function safeApproveAllTokens()
+        external
+        override
+        onlyGovernor
+        nonReentrant
+    {
+        // to add liquidity to the clPool
+        IERC20(WETH).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(positionManager), type(uint256).max);
+        // to be able to rebalance using the swapRouter
+        IERC20(WETH).safeApprove(address(swapRouter), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(swapRouter), type(uint256).max);
+    }
+
+    /***************************************
+            Balances and Fees
+    ****************************************/
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        require(_asset == WETH, "Only WETH supported");
+
+        // we could in theory deposit to the strategy and forget to call rebalance in the same
+        // governance transaction batch. In that case the WETH that is on the strategy contract
+        // also needs to be accounted for.
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        // just paranoia check, in case there is OETHb in the strategy that for some reason hasn't
+        // been burned yet.
+        uint256 _oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        return underlyingAssets + _wethBalance + _oethbBalance;
+    }
+
+    /**
+     * @dev Returns the balance of both tokens in a given position (excluding fees)
+     * @return _amountWeth Amount of WETH in position
+     * @return _amountOethb Amount of OETHb in position
+     */
+    function getPositionPrincipal()
+        public
+        view
+        returns (uint256 _amountWeth, uint256 _amountOethb)
+    {
+        if (tokenId == 0) {
+            return (0, 0);
+        }
+
+        uint160 _sqrtRatioX96 = getPoolX96Price();
+        (_amountWeth, _amountOethb) = helper.principal(
+            positionManager,
+            tokenId,
+            _sqrtRatioX96
+        );
+    }
+
+    /**
+     * @notice Returns the current pool price in X96 format
+     * @return _sqrtRatioX96 Pool price
+     */
+    function getPoolX96Price() public view returns (uint160 _sqrtRatioX96) {
+        (_sqrtRatioX96, , , , , ) = clPool.slot0();
+    }
+
+    /**
+     * @notice Returns the current active trading tick of the underlying pool
+     * @return _currentTick Current pool trading tick
+     */
+    function getCurrentTradingTick()
+        internal
```

Comment:
nit: Is this supposed to be public? If not, can you rename this to `_getCurrentTradingTick` instead?

Reply:
I think this is a nice convenience function similar to `getPoolX96Price`. They both query slot0() on the clPool and that returns a struct. IMO might be more useful if we want to fetch the state of the underlying pool.

Though my position isn't strong on this issue.

Reply:
Sorry, I should've worded it correctly. It has `internal` keyword and not `public`. My question was about that. Based on your reply, I believe it should be `public` as well

Reply:
actually this is my mistake not reading carefully. Changed to public: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/97437097af6cedae6a16bc0a92d646f1c598303f